test/rbi/rbs_printer_test.rb

```diff
@@ -0,0 +1,956 @@
+        def foo; end
+      RBI
+
+      # Modifiers are ignored in RBS, but we generate comments for them
+      # we ignore the checked level
+      assert_equal(<<~RBI, rbi.rbs_string)
+        # @abstract
+        # @override
+        # @overridable
+        def foo: -> void
+      RBI
+    end
+
+    def test_print_methods_with_signature_with_type_parameters
+      rbi = parse_rbi(<<~RBI)
+        sig { type_parameters(:U, :V).params(a: T.type_parameter(:U)).returns(T.type_parameter(:V)) }
+        def foo(a); end
+      RBI
+
+      # To avoid conflict with existing constants, we prefix type parameters with `TYPE_`
+      assert_equal(<<~RBI, rbi.rbs_string)
+        def foo: [TYPE_U, TYPE_V] (TYPE_U a) -> TYPE_V
+      RBI
+    end
+
+    def test_print_methods_with_signatures_and_comments
+      rbi = parse_rbi(<<~RBI)
+        # This is a
+        # multiline Comment
+        def foo; end
+
+        # This is a single line comment
+        sig { returns(void) }
+        def bar; end
+
+        # This is a
+        # multiline Comment
+        sig { returns(A) }
+        sig { returns(B) }
+        def baz; end
+      RBI
+
+      assert_equal(<<~RBI, rbi.rbs_string)
+        # This is a
+        # multiline Comment
+        def foo: (*untyped) -> untyped
+
+        # This is a single line comment
+        def bar: -> void
+
+        # This is a
+        # multiline Comment
+        def baz: -> A
+               | -> B
+      RBI
+    end
+
+    def test_print_methods_with_blocks
+      rbi = parse_rbi(<<~RBI)
+        sig { params(block: T.proc.void).returns(R) }
+        sig { params(block: T.untyped).returns(R) }
+        sig { params(block: Proc).returns(R) }
+        sig { params(block: T.nilable(Proc)).returns(R) }
+        sig { params(block: NilClass).returns(R) }
+        def foo(&block); end
+      RBI
+
+      assert_equal(<<~RBI, rbi.rbs_string)
+        def foo: { -> void } -> R
+               | ?{ (?) -> untyped } -> R
+               | { (?) -> untyped } -> R
+               | ?{ (?) -> untyped } -> R
+               | -> R
+      RBI
+    end
+
+    def test_print_method_sig
+      rbi_def = Method.new("foo") do |node|
+        node.params << ReqParam.new("a")
+        node.params << OptParam.new("b", "42")
+        node.params << RestParam.new("c")
+        node.params << KwParam.new("e")
+        node.params << KwOptParam.new("d", "42")
+        node.params << KwRestParam.new("f")
+        node.params << BlockParam.new("g")
+      end
+
+      rbi_sig = Sig.new do |sig|
+        sig.params << SigParam.new("a", "String")
+        sig.params << SigParam.new("b", "Integer")
+        sig.params << SigParam.new("c", "String")
+        sig.params << SigParam.new("e", "Numeric")
+        sig.params << SigParam.new("d", "Object")
+        sig.params << SigParam.new("f", "T.untyped")
+        sig.params << SigParam.new("g", "T.proc.void")
+
+        sig.return_type = "R"
+      end
```

Comment:
I wanted to emphasis a slight difference between the RBI printer and the RBS printer.

Consider this signature:

```rb
sig { params(a: String).void }
```

Only looking at the signature doesn't tell you if `a` is a positional or keyword parameter. So we need the corresponding method.

This is generally not a concern for someone trying to translate a whole file but this is relevant for someone trying to only translate a signature (as we would do in an hypothetical tool that would take a Ruby file using RBI and spit out the corresponding Ruby file using RBS).

As you can see this test actually calls `RBSPrinter#print_method_sig` rather than the normal `visit`.